cloudify_types/cloudify_types/shared_resource/execute_shared_resource_workflow.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2019 Cloudify Platform Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cloudify import ctx, manager
+from cloudify.exceptions import NonRecoverableError
+from cloudify_rest_client.client import CloudifyClient
+
+from cloudify_types.component.polling import (poll_with_timeout,
+                                              is_all_executions_finished,
+                                              verify_execution_state)
+
+from .constants import SHARED_RESOURCE_TYPE
+
+
+def _verify_shared_resource_node():
+    return ctx.target.node.type == SHARED_RESOURCE_TYPE
+
+
+def execute_shared_resource_workflow(workflow_id,
+                                     parameters,
+                                     client_config,
+                                     wf_timeout,
+                                     redirect_logs):
+    if not _verify_shared_resource_node():
+        raise NonRecoverableError('Tried to execute "{0}" workflow on a non '
+                                  'SharedResource node "{1}"'.format(
+                                    workflow_id, ctx.target.node.id))
+
+    ctx.logger.info("Setting up required input for executing workflow on"
+                    "a SharedResource.")
+    # Cloudify client setup
+    if client_config:
```

Comment:
Should we be getting the local client or the target node client (what if they are not the same)?

Reply:
it's a good idea to take the client definition from the target node (of course if not define set the local one)